apps/www/_blog/2023-04-10-o11y-day.mdx

```diff
@@ -0,0 +1,207 @@
+- **Active cache warming**
+  - Query results are proactively warmed at a configurable interval for a combination of fast response times and as-realtime-as-needed data.
+- **Query sandboxing**
+  - If an Endpoint query contains a CTE and the sandbox option is selected, the Endpoint will inject the query string of the `sql` query parameter into the Endpoint SQL replacing the default query (the part of the SQL query after the CTE).
+  - Endpoints parse SQL to allow `select` queries only. No DML or DDL statements are permitted to run through Logflare Endpoints.
+
+With this feature set, Supabase has been able to build any view we’ve needed on top of billions of daily log events.
+
+### Logflare Endpoint Example
+
+Using webhooks, we can send all GitHub events in the Supabase organization to Logflare. The webhook sends structured events, and Logflare transforms the payload into metadata:
+
+```json
+{
+  "event_message": "supabase/supabase | JohannesBauer97 | created",
+  "id": "0d48b71d-91c5-4356-82c7-fdb299b625d0",
+  "metadata": {
+    "sender": {
+      "id": 15695124,
+      "login": "JohannesBauer97",
+      "node_id": "MDQ6VXNlcjE1Njk1MTI0",
+      "site_admin": false,
+      "type": "User",
+      "url": "https://api.github.com/users/JohannesBauer97"
+    },
+    "starred_at": "2023-03-30T20:33:55Z",
+		...
+  },
+  "timestamp": 1680208436849642
+}
+```
+
+We’re interested in the top contributors, which can be extracted with SQL (in BigQuery dialect):
+
+```sql
+select
+  count(t.timestamp) as count,
+  s.login as gh_user
+from
+  `github.supabase.webhooks` as t
+  cross join unnest(metadata) as m
+  cross join unnest(m.sender) as s
+where
+  timestamp::date > current_date() - @day::int
+group by
+  gh_user
+order by
+  count desc
+limit
+  25
+```
+
+With this view in place, we can use Endpoints to provide an API that we can hit from our application:
+
+```bash
+curl "https://logflare.app/endpoints/query/69425db0-1cfb-48b4-84c7-2a872b6f0a61" \
+ -H 'Content-Type: application/json; charset=utf-8' \
+ -G -d "day=30"
+```
+
+This returns a JSON response with the top org wide contributors for the last 30 days!
+
+```json
+{
+  "result": [
+    {
+      "count": 23404,
+      "gh_user": "vercel[bot]"
+    },
+    {
+      "count": 10005,
+      "gh_user": "joshenlim"
+    },
+    {
+      "count": 7026,
+      "gh_user": "MildTomato"
+    },
+    {
+      "count": 6405,
+      "gh_user": "fsansalvadore"
+    },
+    {
+      "count": 5195,
+      "gh_user": "saltcod"
+    },
+    {
+      "count": 3454,
+      "gh_user": "alaister"
+    },
+    {
+      "count": 2691,
+      "gh_user": "kevcodez"
+    },
+    {
+      "count": 2117,
+      "gh_user": "gregnr"
+    },
+    {
+      "count": 1769,
+      "gh_user": "Ziinc"
+    },
+    {
+      "count": 1749,
+      "gh_user": "chasers"
+    },
+    {
+      "count": 1430,
+      "gh_user": "Isaiah-Hamilton"
+    },
+    ...
```

Comment:
```suggestion
    ...
```

Reply:
if you really want to leave these in then squash them:

```
  "result": [
    { "count": 23404,  "gh_user": "vercel[bot]" },
    { "count": 10005, "gh_user": "joshenlim"  },
    
    etc
```

Reply:
fxied

Reply:
fixed **